File: spoj-br/C/NLOGONIA/NLOGONIA.c
// https://br.spoj.com/problems/NLOGONIA/

#include<stdio.h>
 
int main() 
{
  int K;
    while (scanf("%d", &K) && K) 
	{
       int N, M, coord[K], coord1[K], c;
       scanf("%d %d", &N, &M);
	   
       for (c = 0; c < K; c++)
         scanf("%d %d", &coord[c], &coord1[c]);
       for (c = 0; c < K; c++) 
         if (coord[c] == N || coord1[c] == M)
             printf("divisa\n");
            else if(coord[c] < N && coord1[c] > M)
             printf("NO\n");
			else if(coord[c] > N && coord1[c] > M)
             printf("NE\n");
			else if(coord[c] > N && coord1[c] < M)
             printf("SE\n");
            else if(coord[c] < N && coord1[c] < M)
             printf("SO\n");    
    }
}

File: spoj-br/Swift/NLOGONIA/NLOGONIA.swift
// https://br.spoj.com/problems/NLOGONIA/

// MARK: - Functions
func getK() -> Int {
    return Int(readLine()!)!
}

func getValues() -> [Int] {
    return readLine()!.split(separator: " ").map { Int($0)! }
}

func printPosition(coord: [Int], n: Int, m: Int) {

    let coord1 = coord[0]
    let coord2 = coord[1]

    if (coord1 == n || coord2 == m) {
        print("divisa")
    } else if (coord1 < n && coord2 > m) {
        print("NO")
    } else if (coord1 > n && coord2 > m) {
        print("NE")
    } else if (coord1 > n && coord2 < m) {
        print("SE")
    } else {
        print("SO")
    }
}

// MARK: - Proge
var k = 0
k = getK()
while(k > 0) {

    var n = 0, m = 0
    let values = getValues()
    n = values[0]
    m = values[1]

   for index in 1...k {
        let coord = getValues()
        printPosition(coord: coord, n: n, m: m)
   }
    
    k = getK()
}